### PR TITLE
Vision tech

### PR DESCRIPTION
**Pull request by Iiropel**
First of the year.

### CODE REVIEW

File: MorkoUnityProject/MorkoUnityProject.sln [hunk 1]
```diff
@@ -1,5 +1,5 @@
-﻿Microsoft Visual Studio Solution File, Format Version 11.00
```
Comment (SAMPROO) [inline MorkoUnityProject/MorkoUnityProject.sln:1]: Remove .sln files from merge

File: MorkoUnityProject/Assets/Scripts/Vision/RendTextScaler.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        }
+        cam.targetTexture = new RenderTexture(Screen.width, Screen.height, 24);
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
```
Comment (SAMPROO) [inline MorkoUnityProject/Assets/Scripts/Vision/RendTextScaler.cs:25]: Delete non used Update

File: MorkoUnityProject/Assets/Scripts/Vision/PostFx.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PostFx : MonoBehaviour
+{
+    public Material visionEffectMaterial;
+    public Material visionEffectMaterial2;
```
Comment (SAMPROO) [inline MorkoUnityProject/Assets/Scripts/Vision/PostFx.cs:8]: Not used?

File: MorkoUnityProject/Assets/Scripts/FieldOfView.cs [hunk 4]
```diff
@@ -73,18 +83,53 @@ void DrawFieldOfView()
...
+
+        }
+        fullMesh.Clear();
+        fullMesh.vertices = vertices;
+        fullMesh.triangles = triangles;
+        fullMesh.RecalculateNormals();
+    }
+
+
+    int mask = 1 << 8;
```
Comment (oelnenimmat) [inline MorkoUnityProject/Assets/Scripts/FieldOfView.cs:123]: make const and move to top with other variables

File: MorkoUnityProject/Assets/Scripts/FieldOfView.cs [hunk 5]
```diff
@@ -7,20 +7,29 @@
...
         viewMesh = new Mesh();
         viewMesh.name = "View Mesh";
         viewMeshFilter.mesh = viewMesh;
+
+        fullMesh = new Mesh();
+        fullMesh.name = "Full Mesh";
+        fullMeshFilter.mesh = fullMesh;
     }
 
     public Vector3 DirFromAngle(float degreeAngle, bool globalAngle)
```
Comment (oelnenimmat) [inline MorkoUnityProject/Assets/Scripts/FieldOfView.cs:35]: change `globalAngle` to `useGlobalAngle` or similar to communicate its actual meaning

File: MorkoUnityProject/Assets/Scripts/FieldOfView.cs [hunk 6]
```diff
@@ -73,18 +83,53 @@ void DrawFieldOfView()
         viewMesh.triangles = triangles;
         viewMesh.RecalculateNormals();
     }
+    void DrawFullCone()
+    {
+        int stepCount = Mathf.RoundToInt(viewAngle * meshResolution);
+        float stepAngleSize = viewAngle / stepCount;
+        List<Vector3> viewPoints = new List<Vector3>();
```
Comment (oelnenimmat) [inline MorkoUnityProject/Assets/Scripts/FieldOfView.cs:90]: This can be changed to `Vector3[stepCount]`, since it has fixed size

**Review by oelnenimmat — APPROVED** [review]
Added comments, look into them.

**Pull request merged**